aten/src/TH/generic/THTensorApply.hpp

```diff
@@ -158,7 +158,7 @@ if (std::isnan(val)) break;
 #endif
 
 static inline scalar_t THTensor_(powOne)(scalar_t x, scalar_t y) {
-#if defined(TH_REAL_IS_FLOAT)
+#if defined(TH_REAL_IS_FLOAT) || defined(TH_REAL_IS_HALF)
```

Comment:
I... doubt this actually works?

Reply:
It fixes related test failures and you can find similar conditions all around the place, for example the same file but just couple lines above, there will be 
`#if defined(TH_REAL_IS_FLOAT) || defined(TH_REAL_IS_DOUBLE)`